libraries/net/eth/EthernetInterface/EthernetInterface.cpp

```diff
@@ -59,16 +62,19 @@ static void netif_status_callback(struct netif *netif) {
     }
 }
 
-static void init_netif(ip_addr_t *ipaddr, ip_addr_t *netmask, ip_addr_t *gw) {
+static int init_netif(ip_addr_t *ipaddr, ip_addr_t *netmask, ip_addr_t *gw) {
     tcpip_init(tcpip_init_done, NULL);
     tcpip_inited.wait();
-    
     memset((void*) &netif, 0, sizeof(netif));
-    netif_add(&netif, ipaddr, netmask, gw, NULL, eth_arch_enetif_init, tcpip_input);
+    struct netif * ret;
+    ret = netif_add(&netif, ipaddr, netmask, gw, NULL, eth_arch_enetif_init, tcpip_input);
+    if( ret == NULL) {
```

Comment:
please fix a space -> ``if(ret == NULL)``, use rebase or I'll do it when merging